src/PerformanceTests/Tests/Enabled/CSharp_CA1856.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.CSharp.Analyzers.Performance;
+using PerformanceTests.Utilities;
+using PerfUtilities;
+
+namespace CSharpPerformanceTests.Enabled
+{
+    public class CSharp_CA1856
+    {
+        [IterationSetup]
+        public static void CreateEnvironmentCA1856()
+        {
+            var sources = new List<(string name, string content)>();
+            string attributeSource = @"#nullable enable
+namespace System.Diagnostics.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+    public sealed class ConstantExpectedAttribute : Attribute
+    {
+        public object? Min { get; set; }
+        public object? Max { get; set; }
+    }
+}";
+            sources.Add(("attributeSource", attributeSource));
+
+            for (var i = 0; i < Constants.Number_Of_Code_Files; i++)
+            {
+                var name = "TypeName" + i;
+                sources.Add((name, @$"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+class {name}
+{{
+    public void Test([ConstantExpected(Min = 30, Max = 15)] int val) // diagnostic x2
+    {{
+    }}
+
+    public void Test2([ConstantExpected(Min = 0, Max = 15)] string val) // diagnostic x2
+    {{
+    }}
+
+    public void Test3<T>([ConstantExpected(Min = 0, Max = 15)] T val) // diagnostic
+    {{
+    }}
+
+    public void Test4([ConstantExpected(Min = 0, Max = int.MaxValue)] byte val) // diagnostic
+    {{
+    }}
+
```

Comment:
Perf test doesn't need to cover all scenario, main simple scenario would be enough, I would add one failing one success scenarios
```suggestion
    public void Test2([ConstantExpected(Min = 0, Max = 15)] int val) // no diagnostic
    {{
    }}
```